pkg/lakom/verifysignature/verifier.go

```diff
@@ -134,7 +138,7 @@ func NewCacheVerifier(cache SignatureVerificationResultCache, verifier Verifier)
 // Verify check cosign signature of an image. Firstly it checks if the cache have an entry
 // for the verification state of the image and returns it. If the cache have no entry,
 // it uses the verifier to do the real verification, persists the result in the cache and return it.
-func (r *cacheVerifier) Verify(ctx context.Context, image string, kcr utils.KeyChainReader) (bool, error) {
+func (r *cacheVerifier) Verify(ctx context.Context, image string, kcr utils.KeyChainReader, insecure bool) (bool, error) {
```

Comment:
In theory, the cache verifier should not be aware of the insecure option. It is also not aware of the implementation of the `actualVerifier`, i.e. it could be anything that implements the `Verifier` interface and it might not consider the insecure option. 

I would propose to not add `insecure bool` parameter to the `Verify` method, instead put this feature only in the `directVerifier` and configure it during initialization, just like the cosign public keys are configured. Then only the directVerifier implementation of `Verify` will be aware of the `insecure` option. 

I mean something like:

```go
type directVerifier struct {
	publicKeys []crypto.PublicKey
	insecure   bool
}

// NewDirectVerifier creates new verifier and returns it.
func NewDirectVerifier(keys []crypto.PublicKey, insecure bool) *directVerifier {
	dv := &directVerifier{
		publicKeys: keys,
		insecure:   insecure,
	}
	return dv
}

// Verify check if image is signed by at least one of the configured cosign public keys.
func (r *directVerifier) Verify(ctx context.Context, image string, kcr utils.KeyChainReader) (bool, error) {
	opts := []name.Option{}
	if r.insecure {
		opts = append(opts, name.Insecure)
	}
...
```

Reply:
I agree, your solution is better. I'll apply it